docs/en_US/HowToImplementTrainingService.md

```diff
@@ -0,0 +1,149 @@
+
+__setClusterMetadata(key: string, value: string)__  
+ClusterMetadata is the data related to platform details, for examples, the ClusterMetadata defined in remote machine server is:
+```
+export class RemoteMachineMeta {
+    public readonly ip : string;
+    public readonly port : number;
+    public readonly username : string;
+    public readonly passwd?: string;
+    public readonly sshKeyPath?: string;
+    public readonly passphrase?: string;
+    public gpuSummary : GPUSummary | undefined;
+    /* GPU Reservation info, the key is GPU index, the value is the job id which reserves this GPU*/
+    public gpuReservation : Map<number, string>;
+
+    constructor(ip : string, port : number, username : string, passwd : string, 
+        sshKeyPath : string, passphrase : string) {
+        this.ip = ip;
+        this.port = port;
+        this.username = username;
+        this.passwd = passwd;
+        this.sshKeyPath = sshKeyPath;
+        this.passphrase = passphrase;
+        this.gpuReservation = new Map<number, string>();
+    }
+}
+```
+The metadata includes the host address, the username or other configuration related to the platform. Users need to define their own metadata format, and set the metadata instance in this function. This function is called before the experiment is started to set the configuration of remote machines.
+
+__getClusterMetadata(key: string)__   
+This function will return the metadata value according to the values, it could be left empty if users don't need to use it.
+
+__submitTrialJob(form: JobApplicationForm)__  
+SubmitTrialJob is a function to submit new trial jobs, users should generate a job instance in TrialJobDetail type. TrialJobDetail is defined as follow:
+```
+interface TrialJobDetail {
+    readonly id: string;
+    readonly status: TrialJobStatus;
+    readonly submitTime: number;
+    readonly startTime?: number;
+    readonly endTime?: number;
+    readonly tags?: string[];
+    readonly url?: string;
+    readonly workingDirectory: string;
+    readonly form: JobApplicationForm;
+    readonly sequenceId: number;
+    isEarlyStopped?: boolean;
+}
+```
+According to different kinds of implementation, users could put the job detail into a job queue, and keep  fetching the job from the queue and start preparing and running them. Or they could finish preparing and running process in this function, and return job detail after the submit work.
+
+__cancelTrialJob(trialJobId: string, isEarlyStopped?: boolean)__  
+If this function is called, users should execute cancel command to kill a trial job according to trialJobId.
+
+__updateTrialJob(trialJobId: string, form: JobApplicationForm)__  
+This function is called to update the trial job's status, users should execute the corresponding commands to detect the status of a trial process, and change the trial job's status to `RUNNING`, `SUCCEED`, `FAILED` etc.
```

Comment:
"execute the corresponding commands'   maybe too specific,  user  may want to call os APIs which are not commands.

Reply:
changed to "This function is called to update the trial job's status, trial job's status should be detected according to different platform, and be updated to `RUNNING`, `SUCCEED`, `FAILED` etc."